meteor/client/lib/ShowStyleVariantImportButton.tsx

```diff
@@ -0,0 +1,168 @@
+}
+
+export const ShowStyleVariantImportButton = withTranslation()(
+	class ShowStyleVariantImportButton extends React.Component<
+		IShowStyleVariantImportProps,
+		IShowStyleVariantImportState
+	> {
+		constructor(props: IShowStyleVariantImportProps) {
+			super(props)
+
+			this.state = {
+				timestampedFileKey: Date.now(),
+			}
+		}
+
+		private getDuplicatedShowStyleVariantAmount(showStyleVariant: ShowStyleVariant): number {
+			const importedName = showStyleVariant.name.split(' ')[0]
+			return this.props.showStyleVariants.filter((variant: ShowStyleVariant) => {
+				const existingName = variant.name.split(' ')[0]
+				return existingName === importedName
+			}).length
+		}
+
+		private importShowStyleVariants(event: React.ChangeEvent<HTMLInputElement>): void {
+			const file = event.target.files?.[0]
+			if (!file) {
+				return
+			}
+
+			const reader = new FileReader()
+
+			reader.onload = () => {
+				this.setState({
+					timestampedFileKey: Date.now(),
+				})
+				const fileContents = reader.result as string
+
+				const newShowStyleVariants: ShowStyleVariant[] = []
+				try {
+					JSON.parse(fileContents).map((showStyleVariant: ShowStyleVariant) =>
+						newShowStyleVariants.push(showStyleVariant)
+					)
+					if (!Array.isArray(newShowStyleVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				this.importShowStyleVariantsFromArray(newShowStyleVariants)
+			}
+			reader.readAsText(file)
+		}
+
+		private importShowStyleVariantsFromArray(importedShowStyleVariants: ShowStyleVariant[]): void {
+			importedShowStyleVariants.forEach((showStyleVariant: ShowStyleVariant, index: number) => {
+				const rank = this.props.showStyleVariants.length
+				showStyleVariant._rank = rank + index
+				if (this.showStyleVariantAlreadyExists(showStyleVariant)) {
+					this.provideImportOptions(showStyleVariant)
+				} else {
+					this.importShowStyleVariant(showStyleVariant)
+				}
+			})
+		}
+
+		private showStyleVariantAlreadyExists(importedShowStyleVariant: ShowStyleVariant): boolean {
+			const exists = this.props.showStyleVariants.find(
+				(variant: ShowStyleVariant) => variant.name === importedShowStyleVariant.name
+			)
+			return !!exists
+		}
+
+		private provideImportOptions(showStyleVariant: ShowStyleVariant) {
+			doModalDialog({
+				title: t('Do you want to replace this variant?'),
+				yes: t('Replace'),
+				no: t('Keep both variants'),
+				onAccept: () => {
+					this.replaceShowStyleVariant(showStyleVariant)
+				},
+				onSecondary: () => {
+					const resemblingNameCount = this.getDuplicatedShowStyleVariantAmount(showStyleVariant)
+					showStyleVariant.name += ' (' + resemblingNameCount + ')'
```

Comment:
It would become `MyVariant (2)` due to the search method you've commented on as well. Only the string before a whitespace is used, so the (1) on `MyVariant (1)` is not considered.

Reply:
An idea for comparing instead of taking the string before the first space could be to remove the ` (1)` part.
This can be done with `showStyleName.replace(/ +\(\d+\)$/, '')`.

Reply:
Updated to use the suggestion.